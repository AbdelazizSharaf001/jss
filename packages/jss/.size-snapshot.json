{
  "dist/jss.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  },
  "dist/jss.min.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  }
}
