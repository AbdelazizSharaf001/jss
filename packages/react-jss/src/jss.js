// @flow
import {create} from 'jss'
import preset from 'jss-preset-default'

export const jss = create(preset())
