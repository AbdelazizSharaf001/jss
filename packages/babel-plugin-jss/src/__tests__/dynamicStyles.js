import {stripIndent} from 'common-tags'
import {sheets} from 'jss'
import {transform} from './testUtils'

describe('dynamic styles (with function values or rules)', () => {
  beforeEach(() => {
    sheets.reset()
  })

  test('extract static rule, dynamic rule is an arrow function', () => {
    const code = stripIndent`
      import jss from 'jss';
      createStyleSheet({
        a: {
          color: 'red'
        },
        b: {
          color: () => {}
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('extract static rule, dynamic rule is a function', () => {
    const code = stripIndent`
      import jss from 'jss';
      createStyleSheet({
        a: {
          color: 'red'
        },
        b: {
          color: function () {}
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('extract static rule, dynamic rule is a ref', () => {
    const code = stripIndent`
      import jss from 'jss';

      function f() {}

      createStyleSheet({
        a: {
          color: 'red'
        },
        b: {
          color: f
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('extract static properties from mixed rules', () => {
    const code = stripIndent`
      import jss from 'jss';
      createStyleSheet({
        a: {
          color: 'red',
          width: () => {}
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })
})
