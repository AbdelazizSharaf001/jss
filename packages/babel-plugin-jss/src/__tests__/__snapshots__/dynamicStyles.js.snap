// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic styles (with function values or rules) extract static properties from mixed rules 1`] = `
"[v6]
import jss from 'jss';
createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}',

  a: {
    width: () => {}
  }
}, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import jss from 'jss';
createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\",
  a: {
    width: () => {}
  }
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`dynamic styles (with function values or rules) extract static rule, dynamic rule is a function 1`] = `
"[v6]
import jss from 'jss';
createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}',

  b: {
    color: function () {}
  }
}, {
  'classes': {
    'a': 'a-id',
    'b': 'b-id'
  }
});
[v7]
import jss from 'jss';
createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\",
  b: {
    color: function () {}
  }
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\",
    \\"b\\": \\"b-id\\"
  }
});"
`;

exports[`dynamic styles (with function values or rules) extract static rule, dynamic rule is a ref 1`] = `
"[v6]
import jss from 'jss';

function f() {}

createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}',

  b: {
    color: f
  }
}, {
  'classes': {
    'a': 'a-id',
    'b': 'b-id'
  }
});
[v7]
import jss from 'jss';

function f() {}

createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\",
  b: {
    color: f
  }
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\",
    \\"b\\": \\"b-id\\"
  }
});"
`;

exports[`dynamic styles (with function values or rules) extract static rule, dynamic rule is an arrow function 1`] = `
"[v6]
import jss from 'jss';
createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}',

  b: {
    color: () => {}
  }
}, {
  'classes': {
    'a': 'a-id',
    'b': 'b-id'
  }
});
[v7]
import jss from 'jss';
createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\",
  b: {
    color: () => {}
  }
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\",
    \\"b\\": \\"b-id\\"
  }
});"
`;
